src/sigmatcher/definitions.py

```diff
@@ -49,6 +67,39 @@ def check_strings(self, strings: List[str]) -> List[str]:
     def capture(self, value: str) -> List[str]:
         return self.signature.findall(value)
 
+    def get_dependencies(self) -> List[str]:
+        return [macro.rpartition(".")[0] for macro in self._get_raw_macros]
+
+    @cached_property
+    def _get_raw_macros(self) -> Set[str]:
+        return set(self.MACRO_REGEX.findall(self.signature.pattern))
+
+    def resolve_macros(self, results: Dict[str, Union[Result, Exception, None]]) -> Self:
+        if not self._get_raw_macros:
+            return self
+
+        new_pattern = self.signature.pattern
+        for macro in self._get_raw_macros:
```

Comment:
Resolving a single macro should be common to all the kinds signatures, so refactor the loop body into a method in `BaseSignature`

Reply:
I prefer to first implement other signature types, and only then refactor common code, because I am still unsure about having the same macro syntax / format for all of them.

Reply:
Alright.  
But the syntax (pattern -> internal macro object) doesn't have to be common for you to share the macro resolution logic (internal macro object -> string)

Reply:
Why are you not letting me be lazy :cry: